libfwupd/fwupd-release.c

```diff
@@ -545,36 +545,82 @@ fwupd_release_get_metadata_item (FwupdRelease *release, const gchar *key)
  * fwupd_release_get_uri:
  * @release: A #FwupdRelease
  *
- * Gets the update uri.
+ * Gets the default update URI.
  *
- * Returns: the update uri, or %NULL if unset
+ * Returns: the update URI, or %NULL if unset
  *
  * Since: 0.9.3
+ * Deprecated: 1.5.6: Use fwupd_release_get_locations() instead.
  **/
 const gchar *
 fwupd_release_get_uri (FwupdRelease *release)
 {
 	FwupdReleasePrivate *priv = GET_PRIVATE (release);
 	g_return_val_if_fail (FWUPD_IS_RELEASE (release), NULL);
-	return priv->uri;
+	if (priv->locations->len == 0)
+		return NULL;
+	return (const gchar *) g_ptr_array_index (priv->locations, 0);
 }
 
 /**
  * fwupd_release_set_uri:
  * @release: A #FwupdRelease
  * @uri: the update URI
  *
- * Sets the update uri, i.e. where you can download the firmware from.
+ * Sets the update URI, i.e. where you can download the firmware from.
  *
  * Since: 0.9.3
+ * Deprecated: 1.5.6: Use fwupd_release_add_location() instead.
  **/
 void
 fwupd_release_set_uri (FwupdRelease *release, const gchar *uri)
+{
+	g_return_if_fail (FWUPD_IS_RELEASE (release));
+	fwupd_release_add_location (release, uri);
```

Comment:
to really be compatible to the old behavior shouldn't this clear the old locations and just set the one?

Reply:
Yes, indeed. I couldn't see why CI failed; any ideas with that very welcome.

Reply:
Doh, CI fixed.